secure/storage/src/vault.rs

```diff
@@ -114,26 +114,38 @@ impl VaultStorage {
     fn set_policy(
         &self,
         policy_name: &str,
+        engine: &VaultEngine,
         key: &str,
         capabilities: &[Capability],
     ) -> Result<(), Error> {
-        let (path, policy_name) = if let Some(namespace) = &self.namespace {
-            let path = format!("secret/data/{}/{}", namespace, key);
-            let policy_name = format!("{}/{}", namespace, policy_name);
-            (path, policy_name)
-        } else {
-            (format!("secret/data/{}", key), policy_name.to_string())
-        };
+        let path = format!("{}/{}", engine.to_policy_path(), self.name(key, engine));
```

Comment:
Nit: it might be clearer to only define this path when it's needed below (e.g., like 148)